src/ReTestItems.jl

```diff
@@ -214,15 +214,27 @@ function _runtests(shouldrun, paths, nworkers::Int, nworker_threads::Int, worker
             return _runtests_in_current_env(shouldrun, paths, proj_file, nworkers, nworker_threads, worker_init_expr, testitem_timeout, retries, verbose_results, debug, report, logs)
         else
             @debugv 1 "Activating test environment for `$proj_file`"
-            return Pkg.activate(proj_file) do
-                TestEnv.activate() do
-                    _runtests_in_current_env(shouldrun, paths, proj_file, nworkers, nworker_threads, worker_init_expr, testitem_timeout, retries, verbose_results, debug, report, logs)
+            orig_proj = Base.active_project()
+            try
+                if haskey(TEST_ENVS, proj_file)
+                    testenv = TEST_ENVS[proj_file]
+                    Base.set_active_project(testenv)
```

Comment:
what happens if the the testenv file has been removed?

Reply:
should we have a `isfile(testenv)` check to be safe?

Reply:
Yeah, good call.